data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/client/PeerForwarderClient.java

```diff
@@ -24,22 +28,24 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
+import static org.opensearch.dataprepper.peerforwarder.PeerForwarderConfiguration.DEFAULT_PEER_FORWARDING_URI;
+
 public class PeerForwarderClient {
     private static final Logger LOG = LoggerFactory.getLogger(PeerForwarderClient.class);
-    // TODO: Update the URI to something similar to "/event/forward"
-    private static final String URI = "/log/ingest";
-    private static final int ASYNC_REQUEST_THREAD_COUNT = 200;
 
-    private final ObjectMapper objectMapper;
+    private final PeerForwarderConfiguration peerForwarderConfiguration;
     private final PeerForwarderClientFactory peerForwarderClientFactory;
+    private final ObjectMapper objectMapper;
     private ExecutorService executorService;
     private PeerClientPool peerClientPool;
 
-    public PeerForwarderClient(final ObjectMapper objectMapper,
-                               final PeerForwarderClientFactory peerForwarderClientFactory) {
-        this.objectMapper = objectMapper;
+    public PeerForwarderClient(final PeerForwarderConfiguration peerForwarderConfiguration,
+                               final PeerForwarderClientFactory peerForwarderClientFactory,
+                               final ObjectMapper objectMapper) {
+        this.peerForwarderConfiguration = peerForwarderConfiguration;
         this.peerForwarderClientFactory = peerForwarderClientFactory;
-        executorService = Executors.newFixedThreadPool(ASYNC_REQUEST_THREAD_COUNT);
+        this.objectMapper = objectMapper;
+        executorService = Executors.newFixedThreadPool(peerForwarderConfiguration.getClientThreadCount());
```

Comment:
I guess it's a better to inject the `PeerForwarderConfiguration` here than writing a public method in `PeerForwarderClientFactory` to get the `PeerForwarderConfiguration`.

Reply:
My main concern is the source of truth splitting in the class itself. But since we are using DI to create singleton that is not an outstanding issue in the whole picture.